File: 0219.contains_duplicate_II/solution.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> storage = new HashMap<>();
        for (int i=0;i<nums.length;i++){
            if (storage.containsKey(nums[i])){
                if ((i - storage.get(nums[i])) <= k){
                    return true;
                }
            }
            storage.put(nums[i], i);
        }
        return false;
    }
}

File: 0219.contains_duplicate_II/solution.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        """ 
        只要有一个重复值的索引差 小于等于 k 即可 
        通过dict存值，出现重复指时，进行比较索引值，如果大于则覆盖原有索引
        """
        storage = {}
        for i, value in enumerate(nums):
            if value in storage:
                if (i - storage[value]) <= k:
                    return True
            storage[value] = i
        return False